src/transformers/pipelines/object_detection.py

```diff
@@ -120,14 +120,14 @@ def __call__(
         images = [self.load_image(image) for image in images]
 
         with torch.no_grad():
-            inputs = self.feature_extractor(images=images, return_tensors="pt")
-            outputs = self.model(**inputs)
-
             if self.framework == "pt":
                 target_sizes = torch.IntTensor([[im.height, im.width] for im in images])
             else:
                 raise ValueError("The ObjectDetectionPipeline is only available in PyTorch.")
```

Comment:
This ValueError will never be rasied, because this is encapsulated in a `with torch.no_grad()`. So if Torch is not available, an error will already be raised above.

Reply:
Also, I think that instead of using self.framework, shouldn't we directly use `is_torch_available`?

Reply:
deleted the `raise` and replaced `self.framework` with `is_torch_available`.
Main purpose of this checl is to address https://github.com/huggingface/transformers/pull/12886#discussion_r698565015
Please let me know if `is_torch_available` is good enough for https://github.com/huggingface/transformers/pull/12886#discussion_r698565015